Cargo.toml

```diff
@@ -86,6 +86,7 @@ iceoryx2-tunnels-end-to-end-testing = { version = "0.6.1", path = "iceoryx2-tunn
 iceoryx2-tunnels-zenoh = { version = "0.6.1", path = "iceoryx2-tunnels/zenoh" }
 
 anyhow = { version = "1.0.86" }
+bincode = { version = "2.0.1", default-features = false, features = ["alloc","serde"] }
```

Comment:
According to their readme, bincode has a minimum required rust version of 1.85 (see: https://github.com/bincode-org/bincode?tab=readme-ov-file#what-is-bincodes-msrv-minimum-supported-rust-version), iceoryx2 has currently 1.81.

From my side it is not a problem to increment the MSRV on our side but there I might user out there which have a hard requirement on a lower version. @elBoberido do you have any insights here?

Reply:
VxWorks currently only support Rust 1.82. But AFAIK the serializer can be changed for VxWorks via the `recommended.rs` file, so it should not be a big deal for VxWorks.

But I'm not sure what else will be affected with a larger MSRV. Would it be possible to use the bincode crate but not update our MSRV? If bincode is either hidden behind a feature flag, e.g. `nostd`, then it could work.

Generally, especially for embedded use cases it would be good to support an older rust compiler.

There is also a "nuklear" option. I just tried to build bincode with Rust 1.81 and it builds fine. We could carry a small patch to downgrade the MSRV for bincode. It seems the main reason to go with 1.85 was to be able to use the Rust Edition 2024 at some point.

Reply:
@elBoberido @elfenpiff I switched it to `postcard`.